src/bin/pg_autoctl/pgctl.c

```diff
@@ -1455,11 +1469,16 @@ pg_ctl_postgres(const char *pg_ctl, const char *pgdata, int pgport,
 	args[argsIndex++] = "-p";
 	args[argsIndex++] = (char *) intToString(pgport).strValue;
 
-	if (!IS_EMPTY_STRING_BUFFER(listen_addresses))
```

Comment:
don't we still need this check too?

Reply:
I don't think we do, no. The idea is to drive the `-h` argument from the extra new `listen` boolean argument, rather than second guessing the intentions of the caller from the listen_addresses string buffer value. We could protect against a caller mistakenly passing in listen to true with an empty string buffer for listen_addresses though. That would be a bug. I have added such a protection now.